### ISSUE TITLE
`semicolon_if_nothing_returned` is triggered on generated code

### ISSUE THREAD

**Issue by nazar-pc** [rust-lang/rust-clippy#17199]
### Summary

In generated code it is often unnecessarily hard to distinguish whether return value is meaningful or not. It often comes from external macros users have no direct control over.

I believe this lint should be ignored for generated code annotated with `#[automatically_derived]` attribute.

### Lint Name

semicolon_if_nothing_returne

### Version

```text
rustc 1.97.0-nightly (20de910db 2026-05-02)
binary: rustc
commit-hash: 20de910db49d3476ccf49ea79a4b22e2b5dface0
commit-date: 2026-05-02
host: x86_64-unknown-linux-gnu
release: 1.97.0-nightly
LLVM version: 22.1.4
```

**Comment by hibjam** [comment]
Hiya, I'd like to work on this :) 

@rustbot claim
